Assets/HoloToolkit/Input/Scripts/InputSources/GamepadInput.cs

```diff
@@ -15,12 +15,12 @@
 namespace HoloToolkit.Unity.InputModule
 {
     /// <summary>
-    /// GamepadInput class maps Xbox gamepad buttons to the GestureRecognizer.
+    /// GamepadInput class maps Xbox GamePad buttons to the GestureRecognizer.
     /// Gamepad button A press and release maps to tap gesture.
     /// Gamepad button A pressed maps to hold started, completed, canceled gesture.
     /// Gamepad button A pressed plus left joystick rotate maps to navigation gesture.
     /// </summary>
-    [Obsolete("Please use GamePadInputSource or XboxControllerInputSource")]
+    [Obsolete("Please use XboxControllerHandlerBase")]
```

Comment:
I frankly don't understand why the complicated new class for something simple like this. So I need to now implement interfaces to handle GamePad which previously just worked?

Reply:
Yes, because if you weren't implementing the interfaces you were not handling the input correctly.

This new class isn't intended to complicate, and this isn't just something trivial to add onto the `InputManager.prefab` that doesn't follow the appropriate implementation patterns.

Reply:
Then GamepadInput.cs cannot be marked obsolete. We have to keep the simple implementation working for everyone.

Reply:
The new, simple implementation is `XboxControllerHandlerBase`

Reply:
For the most part the underlying logic is the same, with the difference here being that we're handling the input properly though the interface.

![image](https://user-images.githubusercontent.com/13334553/33331592-d87cd164-d416-11e7-824e-4147b1d7beba.png)

The important thing here is customization of the `Select` and `Horizontal Navigation Axis` that were hard coded previously.  Now users can map their own with easy dropdowns.  Because of the updated code structure, devs can now inherit directly from this class and implement their own additional logic on top of the standard inputs.